.github/workflows/deploy-website.yaml

```
@@ -35,6 +30,12 @@ jobs:
   deploy:
     if: github.ref == 'refs/heads/master'
     needs: build
+    concurrency:
+      group: pages
+      cancel-in-progress: true
```

Did you also consider removing this setting completely? What's the downside of deploying twice? 

If I understand correctly, what's happening is:
- PR-1 merged => master is built & deployed
- PR-2 is merged => master is built & deployed

Where PR-2 is somehow faster and thus first PR-2 is deployed, after which PR-1 is deployed and thus leading to an outdated website. Additionally, the current setup is cancelling the _build_ across multiple branches/PRs, which then requires one to manually re-run to fix the build.

Now, we're still grouping globally (`group: pages`), and this works because the `if` is on a single branch, but could still lead to issues across multiple branches I think.

An alternative would be moving this up again, but grouping differently. Arguably, if we merge the two PRs (from the above example), there's no need in building twice. Additionally, if you push 2 commits on a branch, we should be able to cancel them as - again - there's no use in building twice.

```suggestion
    concurrency:
      group: ${{ github.ref }}
      cancel-in-progress: true
```

---

Agree with this change, as this is undesirable:

> Additionally, the current setup is cancelling the build across multiple branches/PRs, which then requires one to manually re-run to fix the build.

Based on the linked docs, I'd say that this would be even safer (and be prepared on possible future usages of `concurrency` as well):  `group: ${{ github.workflow }}-${{ github.ref }}`.

---

I'd say that dropping `cancel-in-progress` would be fine as well.

> Did you also consider removing this setting completely?

However,  I'd say that it is nice if you do multiple commits that `pending` builds will get cancelled:

> When a concurrent job or workflow is queued, if another job or workflow using the same concurrency group in the repository is in progress, the queued job or workflow will be pending. Any previously pending job or workflow in the concurrency group will be canceled.

---

Hmm, [this](https://github.com/PicnicSupermarket/error-prone-support/pull/284#discussion_r990669460) was an old comment that I hadn't flushed yet, hence the funny ordering. Anyway, the question is whether we're okay with the PR as-is, or want to add `cancel-in-progress: true` again.